### PR TITLE
dataframe: load values from backend and distaply in better dropdowns

### PR DESCRIPTION
**Pull request by mscolnick**
* add RPC to `mo.ui.dataframe` to load column values. this loads all values instead of at that particular transform or filter. this adds complexity and could be confusing to users.
* refactor the inputs in form.tsx to be more re-usable and use better inputs (selects, multi-select, etc)
* make panels the same height to avoid content shift
* improving some border styling

### CODE REVIEW

File: marimo/_plugins/ui/_impl/dataframes/dataframe.py [hunk 1]
```diff
@@ -109,6 +120,29 @@ def get_dataframe(self, _args: EmptyArgs) -> GetDataFrameResponse:
             row_headers=get_row_headers(self._value),
         )
 
+    def get_column_values(
```
Comment (akshayka) [inline marimo/_plugins/ui/_impl/dataframes/dataframe.py:123]: Nice, very clean! Gives confidence in the RPC framework.

**Comment by akshayka** [comment]
@mscolnick I merge ms/improve-select-inputs because it said merge when ready, but now I see that caused a merge conflict here. Sorry about that

**Review by akshayka — CHANGES_REQUESTED** [review]
Found a bug that causes output to vanish, see video in PR thread

**Comment by akshayka** [comment]
@mscolnick Here's the bug

[filter-in-bug.webm](https://github.com/marimo-team/marimo/assets/1994308/79182289-2712-424c-b33f-fe0ded00fc26)

File: marimo/_plugins/ui/_impl/dataframes/dataframe.py (hunk 1 above)
Comment (mscolnick) [reply marimo/_plugins/ui/_impl/dataframes/dataframe.py:123]: yep, great job here!

**Comment by mscolnick** [comment]
@akshayka, thanks, i knew exactly where to fix it. i saw it before and then regressed it in my final cleanup. should be good for another reviews, thanks!

**Review by akshayka — APPROVED** [review]

**Pull request merged**